vector/README.md

```diff
@@ -1,6 +1,72 @@
 The `vector` package [![Build Status](https://github.com/haskell/vector/workflows/CI/badge.svg)](https://github.com/haskell/vector/actions?query=branch%3Amaster)
 ====================
 
-An efficient implementation of `Int`-indexed arrays (both mutable and immutable), with a powerful loop optimisation framework.
+This package includes various modules that will allow you to 
+work with vectors and use an optimisation framework called [*fusion*](#fusion). 
+In this context, vector is an `Int`-indexed array-like data structure with a simpler 
+API that can contain any Haskell value. Additionally, its equivalence 
+to C-style arrays and optimisation via fusion accelerates vector’s 
+performance and makes it a great alternative to list. By installing this 
+package, you’ll be able to work with [boxed, unboxed, storable, and primitive 
+vectors](#vectors-available-in-the-package) as well as their generic interface.
```

Comment:
```suggestion
An efficient implementation of `Int`-indexed arrays (both mutable and immutable).
This package provides multiple representations: [boxed, unboxed, storable, and primitive 
vectors](#vectors-available-in-the-package) and generic API which is polymorphic in
vector type. It also has powerful optimization framework based on [stream-fusion](#fusion)
which could eliminate intermediate data structures.
```
I think it would be better to cut down this paragraph a bit.

Reply:
Hi, I cut down the intro, but kept the sentence about vector's benefits (I think it's important to highlight them for newcomers)

Reply:
I dropped this: 

> The advantages of vectors include equivalence to C-style arrays and simple interface

because it's not clear what does "equivalence to C-style arrays" even mean?

Reply:
removed it